src/compiler/comptypes.rs

```diff
@@ -14,6 +14,12 @@ use crate::compiler::srcloc::Srcloc;
 #[derive(Clone, Debug)]
 pub struct CompileErr(pub Srcloc, pub String);
 
+impl From<(Srcloc, String)> for CompileErr {
+    fn from(err: (Srcloc, String)) -> Self {
+        CompileErr(err.0, err.1)
+    }
+}
```

Comment:
I think this is all you really need. I don't think you need the `ErrInto` trait.